### PR TITLE
[FIX] website: apply page search on most specific pages only

### PR DESCRIPTION
**Pull request by bso-odoo**
Since [1] when the fuzzy search was introduced on website pages, the filtering on most specific pages only happens at the end of the search operation. Because of this, the search happens on pages that would be excluded anyway, the limit might be wrongly applied and the count might be wrong.

This commit makes the page search begin by keeping only the most specific pages, then performing the actual search within those pages.

Steps to reproduce:
- Install `website` only and drop a search snippet.
- Search "ax".

=> No results found, but the "All results" link is displayed.

[1]: https://github.com/odoo/odoo/commit/9f9c4bb7e40233e633f97c60fb00ae191e9077af

task-3203794

### CODE REVIEW

**Comment by qsm-odoo** [comment]
@robodoo delegate=detrouxdev

File: addons/website/models/website_page.py [hunk 1]
```diff
@@ -284,19 +284,29 @@ def _search_get_detail(self, website, order, options):
...
+        # Cannot rely on the super's _search_fetch because the search must be
+        # performed among the most specific pages only.
+        fields = search_detail['search_fields']
+        base_domain = search_detail['base_domain']
+        domain = self._search_build_domain(base_domain, search, fields, search_detail.get('search_extra'))
+        # TODO In 16.0 do not rely on _filter_duplicate_pages.
+        most_specific_pages = self.env['website'].with_context(_filter_duplicate_pages='url' not in order)._get_website_pages(
+            domain=expression.AND(base_domain), order=order
+        )
+        results = most_specific_pages.sudo().filtered_domain(domain)
```
Comment (detrouxdev) [inline addons/website/models/website_page.py:296]: Could you explain to me why the sudo is needed here? If I'm not wrong, prior to this commit, the sudo was only performed if `requires_sudo` was in the search detail. Why not here?

**Review by detrouxdev — COMMENTED** [review]
Only a single question, other than that, I don't have much to say, good work! 😄

File: addons/website/models/website_page.py (hunk 1 above)
Comment (bso-odoo) [reply addons/website/models/website_page.py]: Ultimately removed that `sudo` because the result of `_get_website_pages` is already `sudo`.

**Comment by detrouxdev** [comment]
@robodoo r+

**Pull request closed without merging**